### DISCUSSION TITLE
Files saves are named the previous detected video, not the current video

### DISCUSSION THREAD

**Discussion by naetimus** [aclap-dev/video-downloadhelper#1225] [Help and Q&A]
### Your question

I have a problem where files are automatically named whatever the name of the previously detected file is named. I can avoid this by clearing the download history before each time I go to a new URL so there is no previous history to use. Or, I can reload the page so it detects the current video twice in a row. But, other than that, it always names the saved file the wrong name.
Does anyone else have this problem? Any suggestions on what to do?

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"5e3c49a619704606"}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "download_directory": "/Users/####/Movies/Downloads",
    "concurrent_downloads_max": 4,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 31,
    "successfull_dl": 7,
    "history_limit_in_days": 2
}

**Comment by swren7** [comment]
I am facing the same issue.  The titles of downloaded videos are always the title of the previously played video. Looks like it is a array index issue.

**Comment by GrampaWildWilly** [comment]
You might want to read #1132.
